frontend/vre/main.js

```diff
@@ -40,35 +42,41 @@ const SRUIDS = ['hpb', 'vd16', 'vd17', 'vd18', 'gallica', 'cerl-thesaurus'];
 
 var VRERouter = Backbone.Router.extend({
     routes: {
-        ':id/': 'showDatabase',
+        'collection/:id/': 'showCollection',
+        'catalog/:id/': 'showCatalog',
     },
-    showDatabase: function(id) {
+    showCollection: function(id) {
         GlobalVariables.searchView.source = id;
         GlobalVariables.searchView.render();
-        if (SRUIDS.includes(id)) {
-            //$('#content').empty();
-            var sruView = new SRUView();
-            GlobalVariables.searchView.$el.appendTo(sruView.$('.page-header'));
-            $('#content').replaceWith(sruView.$el);
-            var advancedSearchView = new AdvancedSearchView();
-            advancedSearchView.render();
-            GlobalVariables.searchView.listenTo(advancedSearchView, 'fill', GlobalVariables.searchView.fill);
-        }
-        else {
-            // We are not on the HPB search page, so display the
-            // records in the current collection.
-            $('#HPB-info').hide();
-            GlobalVariables.currentVRECollection = GlobalVariables.myCollections.get(id);
-            var collectionView = new CollectionView({model:GlobalVariables.currentVRECollection});
-            GlobalVariables.searchView.$el.appendTo(collectionView.$('.page-header'));
-            $('#content').replaceWith(collectionView.$el);
-            GlobalVariables.records = GlobalVariables.currentVRECollection.getRecords();
-            GlobalVariables.recordsList.remove();
-            GlobalVariables.recordsList = new RecordListManagingView({
-                collection: GlobalVariables.records,
-            });
-            GlobalVariables.recordsList.render().$el.insertAfter($('.page-header'));
-        }
+        // We are not on the HPB search page, so display the
+        // records in the current collection.
+        $('#HPB-info').hide();
+        GlobalVariables.currentVRECollection = GlobalVariables.myCollections.get(id);
+        GlobalVariables.currentCatalog = null;
+        GlobalVariables.collectionDropdown.render();
+        GlobalVariables.catalogDropdown.render();
```

Comment:
This is the approach that was already taken (so you are 100% forgiven for following it): change some global state, then manually trigger a rerender. It works, but isn't elegant and will grow more difficult to manage for every aditional place in which it is applied.

The more elegant and scalable way to manage this kind of thing is to track the currently selected catalog or collection with a model. The model emits an event on every change, which enables the previously selected catalog or collection, the newly selected catalog or collection and all views involved to respond.

It is probably easiest if I show this in a follow-up pull request, so this is definitely not a condition for acceptance. However, you can have a go at it yourself first if you prefer that, or we can do it in a pair programming session.

Reply:
Agree, I felt unease when writing this code! I would not have thought of using a model for this, great! I will create an issue for this now.